tiledb/bioimg/converters/scale.py

```diff
@@ -0,0 +1,272 @@
+
+                tiles.append(tuple(tile))
+
+            self._resolutions.append(tuple(shape))
+            self._tiles.append(tiles)
+
+    @property
+    def resolutions(self) -> List[Tuple[int, ...]]:
+        return self._resolutions
+
+    @property
+    def tiles(self) -> List[List[Tuple[slice, ...]]]:
+        return self._tiles
+
+    def apply(
+        self,
+        input_data: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        level: int,
+        method: ScaleMethod,
+    ) -> None:
+        if input_data.shape != self._initial_shape:
+            raise ValueError(
+                f"Incompatible input shape. Expected {self._initial_shape}, got {input_data.shape}"
+            )
+
+        if level >= len(self._resolutions):
+            raise ValueError(
+                f"Unknown zoom level. Expected 0 to {len(self._resolutions) - 1}, got {level}"
+            )
+
+        out_data[:] = resize(
+            input_data[:],
+            self._resolutions[level],
+            preserve_range=True,
+            order=method.value,
+            anti_aliasing=True,
+        )
+
+    def apply_progressive(
+        self,
+        input_data: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        method: ScaleMethod,
+    ) -> None:
+
+        level = (
+            0
+            if input_data.shape == self._initial_shape
+            else self.resolutions.index(input_data.shape) + 1
+        )
+
+        if level >= len(self._resolutions):
+            raise ValueError(
+                f"Unknown zoom level. Expected 0 to {len(self._resolutions) - 1}, got {level}"
+            )
+
+        out_data[:] = resize(
+            input_data[:],
+            self._resolutions[level],
+            preserve_range=True,
+            order=method.value,
+            anti_aliasing=True,
+        )
+
+    def apply_chunked(
+        self,
+        base: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        level: int,
+        method: ScaleMethod,
+    ) -> None:
+        if base.shape != self._initial_shape:
+            raise ValueError(
+                f"Incompatible input shape. Expected {self._initial_shape}, got {base.shape}"
+            )
+
+        if level >= len(self._resolutions):
+            raise ValueError(
+                f"Unknown zoom level. Expected 0 to {len(self._resolutions) - 1}, got {level}"
+            )
+
+        with Pool(multiprocessing.cpu_count()) as pool:
+            pool.map(
+                partial(
+                    self._scale,
+                    base=base,
+                    out_data=out_data,
+                    level=level,
+                    scale_method=method.value,
+                ),
+                iter_tiles(out_data.domain),
+            )
+
+    def apply_chunked_progressive(
+        self,
+        base: tiledb.array.DenseArray,
+        out_data: tiledb.array.DenseArray,
+        method: ScaleMethod,
```

Comment:
Please pick a single name for the same thing and use it consistently: `base` is called `input_data` in `apply`, `method` is called `scale_method` or `order` elsewhere